### PR TITLE
docs(TASK-5214): update type definitions for useCache

### PR DESCRIPTION
**Pull request by ralphiz**
### Description
Updates to the type definitions for useCache

### Check List
If not relevant to pull request, check off as complete

- [x] All tests passing
- [x] Docs updated with any changes or examples if applicable
- [x] Added tests to ensure new feature(s) work properly

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -465,51 +465,86 @@ export function useRead(args: unknown): any;
...
+ * > ```
  * @param alias string
  * @param fields keys of doc
  * @returns Select keys from doc
  */
-export function useCache<Doc extends PathId>(
+export function useCache<Doc extends PathId, K extends keyof Doc>(
   alias: string,
   fields: K[],
 ): Pick<Doc, K>[] | undefined;
```
Comment (puppybits) [inline index.d.ts:550]: For this to work it needs a function skeleton at the end. 
```ts
export function useCache(args: unknown): any;
```

Also write out the useCache function is a random ts file and make sure the type hints change properly when you change inputs and generics.

**Review by puppybits — CHANGES_REQUESTED** [review]
A couple of notes

File: index.d.ts (hunk 1 above)
Comment (ralphiz) [reply index.d.ts:565]: Awesome, just tried it out using your scratchpad and they show up. Here's a screen recording of my hovering over them on VSCode: https://recordit.co/63hlv9OjD3

Comment (ralphiz) [reply index.d.ts:565]: Also, not sure why build is failing. Should I try and fix it before we merge this PR?

**Pull request merged**